tests/datatypes/test_list.py

```diff
@@ -184,3 +176,88 @@ class InplaceListOperationTests(InplaceOperationTestCase, TranspileTestCase):
 
         'test_modulo_complex',
     ]
+
+class ListComparisonTests(TranspileTestCase):
+    """
+    Tests for List [<, <=, >, >=] List
+    """
+    operators = ['>', '>=', '<', '<=']
+    # comparisons = ''
+    # for o in operators:
+    #     comparisons += '\nprint(x {} y)\n'.format(o)
+    #
+    # comparisons = "\n".join(['print(x {} y)'.format(o) for o in operators])
+
+    comparisons = """
+    print(x > y)
+    print(x >= y)
+    print(x < y)
+    print(x <= y)
+    """
+
+    def test_this_empty(self):
+        """
+        `this` (left list) is empty.
+        """
+
+        test_str = """
+        print('>>> x = []')
+        print('>>> y = [1,2,3]')
+        x = []
+        y = [1,2,3]
+        """
+        self.assertCodeExecution(test_str+self.comparisons)
```

Comment:
You've got the right idea here; my only comment is about the use of the print statements. 

There are two reasons to have a print statement in the test - firstly, to validate correct behavior. This is what the "print(x > y)" is for - you're running the comparison, and making sure you get back the right answer.

The second reason is an an annotation to help debug what is going on. I'm guessing you've taken the lead from the automated tests here; they output `>>> x = ...` as the test runs. However, the key thing is *when* it does this.  The automated tests effectively output "what am I about to do", followed by "what was the result". This means that when a problem occurs, and the output doesn't match, the diff has a clear line indicating what operation was the problem. 

Based on the test you've got here, if everything works as expected, it will work fine, and keep validating that `<`/`>`/`<=`/`>=` behavior works. However, lets say some bug gets introduced and `>` stops working for some reason. Your test will output:
```
>>> x = []
>>> y = [1,2,3]
+False
-True
False
True
True
```

which isn't especially illuminating. However, if it said:
```
>>> x = []
>>> y = [1,2,3]
>>> x > y
+False
-True
>>> x >= y
False
>>> x < y
True
>>> x <= y
True
```
you can tell straight away that the problem is the > operator.

So, if you can add in some output to the template to provide these extra debug helpers, you're got it!